rclcpp/include/rclcpp/node.hpp

```diff
@@ -170,7 +171,23 @@ class Node : public std::enable_shared_from_this<Node>
     const rmw_qos_profile_t & qos_profile = rmw_qos_profile_default,
     std::shared_ptr<Alloc> allocator = nullptr);
 
-  /// Create and return a Subscription.
+  /// Create and return a Publisher.
+  /**
+   * \param[in] topic_name The topic for this publisher to publish on.
+   * \param[in] options Additional options for the created Publisher
+   * \return Shared pointer to the created publisher.
+   */
+  template<
+    typename MessageT,
+    typename Alloc = std::allocator<void>,
+    typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
+  std::shared_ptr<PublisherT>
+  create_publisher(
+    const std::string & topic_name,
+    const PublisherOptions<Alloc> & options,
```

Comment:
This should have a default argument, right?

Reply:
Also, I think we ought to keep the `size_t qos_history_depth` as a separate, non-default argument in this method.

This is related to another issue, about whether or not we should always force users to specify the history depth. But having a another signature that doesn't require you to specify it is a step backwards I think.

Reply:
> This should have a default argument, right?

It couldn't be defaulted while keeping the other signatures, since it's `create_publisher(topic_name)` becomes ambiguous in that case. 

I'll go ahead with the analysis on which other packages need to be changed, which informs how this looks anyways.

> we ought to keep the size_t qos_history_depth as a separate, non-default argument in this method

Ok - I wasn't aware before that we wanted to force users to always specify. I will change in that direction

Reply:
@wjwwood one question for forcing qos_history_depth: which should have precedence, the function argument or the value in the qos_profile in the Options struct?

Reply:
The function argument.

Later, I may have a QoS factory function that requires you to specify the depth to get a QoS, and then make the QoS a required argument, but for now, requiring the depth and overriding the value in the default QoS structure is the way to go.